src/treectl.c

```diff
@@ -2872,20 +2895,28 @@ TreeControlWndProc(
       // Are we over a drop-able sink?
       //
       if (wParam) {
-         if (GetKeyState(VK_CONTROL) < 0)     // CTRL
-            bCopy = TRUE;
-            else if (GetKeyState(VK_MENU)<0 || GetKeyState(VK_SHIFT)<0) // ALT || SHIFT
-               bCopy = FALSE;
+         if (GetKeyState(VK_CONTROL) < 0) {
+            iOperation = DROP_COPY;
+            if (GetKeyState(VK_SHIFT) < 0) {
+               iOperation = DROP_LINK;
+               if (GetKeyState(VK_MENU) < 0) {
+                  iOperation = DROP_JUNC;
+               }
+            }
+         }
+         else if (GetKeyState(VK_MENU) < 0 || GetKeyState(VK_SHIFT) < 0) 
+            // ALT || SHIFT forces a move even if not on same drive
+            iOperation = DROP_MOVE;
             else
-               bCopy = (GetDrive(lpds->hwndSink, lpds->ptDrop) != GetDrive(lpds->hwndSource, lpds->ptDrop));
+            iOperation = (GetDrive(lpds->hwndSink, lpds->ptDrop) != GetDrive(lpds->hwndSource, lpds->ptDrop));

```

Comment:
This depends on DROP_MOVE == 0 and DROP_COPY == 1.

Reply:
The comparison between the two expression still renders into iOperation being either FALSE or TRUE , which now is 
DROP_MOVE == 0 or DROP_COPY == 1. So the result is the same.

That's why I chose the value for DROP_MOVE == 0 and DROP_COPY == 1 so that it is compatible.